src/test/groovy/pl/allegro/tech/discovery/consul/recipes/watch/ConsulWatcherTest.groovy

```diff
@@ -150,6 +150,55 @@ class ConsulWatcherTest extends Specification {
         await().atMost(Duration.FIVE_SECONDS).until({ consumedMessages == ['123', '126'] })
     }
 
+    def "should discard message and reset index if it went backwards"() {
+        given:
+        consul.stubFor(get(urlPathEqualTo('/endpoint'))
+                .inScenario("index_backwards")
+                .whenScenarioStateIs(Scenario.STARTED)
+                .withQueryParam('index', equalTo('0'))
+                .withQueryParam('wait', equalTo('5m'))
+                .willReturn(aResponse()
+                        .withHeader('X-Consul-Index', '123')
+                        .withBody('123')))
+
+        consul.stubFor(get(urlPathEqualTo('/endpoint'))
+                .inScenario("index backwards")
+                .whenScenarioStateIs(Scenario.STARTED)
+                .withQueryParam('index', equalTo('123'))
+                .withQueryParam('wait', equalTo('5m'))
+                .willReturn(aResponse()
+                        .withHeader('X-Consul-Index', '120')
```

Comment:
u can extract X-Consul-Index as a constant

Reply:
I don't believe this will improve readability.   any other votes for?

Reply:
in general it is a good practice to extract string to meaningful constant i.e. `private final static String CONSUL_INDEX_HEADER = "X-Consul-Index"` if it occur more than once or it is not self explaining (i.e. magic numbers)

Reply:
this sounds like a larger change, unrelated to this PR.   if you feel strongly about this, open a new issue/PR :)

Reply:
[malfoj89](https://github.com/malfoj89) - quoting from misattached comment :

> I only meant for those 2 strings here
> 
> Also I don’t feel that should be separate pr to extract two strings

if you look at how other tests are written,  we use inline strings for headers and scenario names.  so I'm not willing to take this, because it would result in inconsistent test structure.  the proper way forward may be to refactor ConsulWatcherTest.groovy (a lot could be done to dedup/simplify it), but I'm not sure it's worth the effort.

Reply:
doing something cuz someone else already did it that way is imho not the way to do things, I'd say if we can't agree on that, let others decide and it will be fine

Reply:
https://www.oreilly.com/library/view/97-things-every/9780596809515/ch08.html

Reply:
to be more specific: https://github.com/allegro/consul-recipes/pull/31 <- this is how I see it